src/eradiate/experiments/_core.py

```diff
@@ -478,11 +479,30 @@ def process(
 
         for ctx_index, spectral_group_dict in mi_results.items():
             for sensor_id, mi_bitmap in spectral_group_dict.items():
-                measure = sensor_to_measure[sensor_id]
-                measure.mi_results[ctx_index] = {
-                    "bitmap": mi_bitmap,
-                    "spp": spp if spp > 0 else measure.spp,
-                }
+                
+                if self.integrator.moment :
+                    split = mi_bitmap.split()
+                    
+                    m2 = split[1][1]
+
+                    # convert to single channel
```

Comment:
Why do we need to do this conditionally? For sure, with a single component, it wouldn't do much, but that would simplify a bit the control flow.

Reply:
Do you mean remove the check for more than one channel altogether? I have removed it.